internal/core/data/v2/application/event.go

```diff
@@ -199,3 +222,49 @@ func EventsByDeviceName(offset int, limit int, name string, dic *di.Container) (
 	}
 	return events, nil
 }
+
+// AsyncDeleteEvents Handles the deletion of device events asynchronously. This function is expected to be running
+// in a go-routine and works with the "DeleteEventsByIds" function for better performance.
+func AsyncDeleteEvents(dic *di.Container, deleteEventsByIdChannel chan []string) {
```

Comment:
channel should be local to this package, initialized by this function before the for loop, not passed in.

Also the ctx.Done needs to be checked in the select to exit when SIGTERM and the Wait group needs to be incremented at the start and decremented when ctx.Done seen. The wg.add(1) need to be called where the `go` statement is used and the a deferred wg.Done() in this func. ctx & wg need to be parameters passed from init(). 
Similar to this:https://github.com/edgexfoundry/edgex-go/blob/70c918de78a758e1281f9e18824a6a08fc9647aa/internal/core/data/init.go#L113

Same for other async func

Reply:
Please ignore an focus on my lest recommendation.